tests/test_docstring_options.cpp

```diff
@@ -82,28 +82,26 @@ TEST_SUBMODULE(docstring_options, m) {
         py::class_<DocstringTestFoo> c(m, "DocstringTestFoo", "This is a class docstring");
 
         c.def_property("value_prop1",
-                          &DocstringTestFoo::getValue,
-                          &DocstringTestFoo::setValue,
-                          "This is a property docstring");
+                       &DocstringTestFoo::getValue,
+                       &DocstringTestFoo::setValue,
+                       "This is a property docstring");
 
         {
             py::options nested_options;
             nested_options.enable_user_defined_docstrings();
 
             c.def_property("value_prop2",
-                          &DocstringTestFoo::getValue,
-                          &DocstringTestFoo::setValue,
-                          "This is a property docstring");
+                           &DocstringTestFoo::getValue,
+                           &DocstringTestFoo::setValue,
+                           "This is a property docstring");
 
-            c.def_property("value_prop3",
-                          &DocstringTestFoo::getValue,
-                          &DocstringTestFoo::setValue);
+            c.def_property(
+                "value_prop3", &DocstringTestFoo::getValue, &DocstringTestFoo::setValue);
 
             options.disable_function_signatures();
 
-            c.def_property("value_prop4",
-                          &DocstringTestFoo::getValue,
-                          &DocstringTestFoo::setValue);
+            c.def_property(
```

Comment:
What about static properties? Any ideas on how to support those as well?

Reply:
There are multiple issues for this (#3815, #2900). I looked into it.

Expectation: The docstring of static property `FooClass.bar_prop` is shown in `help(FooClass)` and `help(FooClass.bar_prop)`.

Problem:
Both `help` calls read the documentation from `FooClass.bar_prop.__doc__`. However, `FooClass.bar_prop` is already the value of the property, so `__doc__` refers to the class of the value, not to the property.
So fundamentally this is as impossible as adding docstrings to attributes. (See rejected PEP 224: https://legacy.python.org/dev/peps/pep-0224/ )

Possible work-around:
It could be done like for enums. List the static properties in the documentation of the parent class:
https://github.com/pybind/pybind11/blob/master/include/pybind11/pybind11.h#L1961
Not sure whether this would look any good.